### ISSUE TITLE
wrong content-type when sharing image as attachment from whatsapp

### ISSUE THREAD

**Issue by lkwg82** [k9mail/k-9#1221]
### Expected behaviour

I expect to have an email with the correct content-type of the attachment. The image should be viewable.

Snippet from the raw email, for example
```
----_com.android.email_66919854356700
Content-Type: image/jpeg;
 name="IMG-20160327-WA0004.jpg"
Content-Transfer-Encoding: base64
Content-Disposition: attachment;
 filename="IMG-20160327-WA0004.jpg";
 size=144895
```

I checked on another phone with no k9.

### Actual behaviour

wrong content-type
```
------E63KOBT6S2CMWF65RZO58S1RLOK4XK
Content-Type: message/rfc822;
 name="IMG-20160327-WA0000.jpg"
Content-Disposition: attachment;
 filename="IMG-20160327-WA0000.jpg";
 size=270447
Content-Transfer-Encoding: 7bit
```

### Steps to reproduce
1. open image in whatsapp
2. share with k9 (email + send to self)
3. open this email and get offered another email programm for viewing the image

### Environment
K-9 Mail version: 5.008

Android version: 6.0.1

Account type: IMAP/SMTP

**Comment by philipwhiuk** [comment]
Fixed by #776

**Comment by cketti** [comment]
I failed to backport this to 5.008. It'll be fixed in 5.009.
